src/backend/tests/projects.test.ts

```diff
@@ -666,4 +667,47 @@ describe('Projects', () => {
       expect(prisma.material_Type.create).toBeCalledTimes(1);
     });
   });
+
+  describe('assigning material assemblies', () => {
+    test('assignment fails because of permissions', async () => {
+      vi.spyOn(prisma.material, 'findUnique').mockResolvedValue(prismaMaterial1);
+      vi.spyOn(prisma.assembly, 'findUnique').mockResolvedValue(prismaAssembly1);
+      const project = { ...prismaProject1, teams: [{ members: [aquaman], leads: [superman] }] };
+      vi.spyOn(prisma.project, 'findFirst').mockResolvedValue(project);
+
+      await expect(ProjectsService.assignMaterialAssembly(theVisitor, 'mid', 'aid')).rejects.toThrow(
+        new AccessDeniedException('Only leadership or above can create a material type')
+      );
+    });
```

Comment:
I'm not sure if this is a problem with my machine, but the test was only working if I manually created a team object and set the members and leads (line 675).

I was using the teams defined in the test data, but although they have valid users in the members and leads lists, they were undefined in the tests. I confirmed this by console.logging them

Reply:
@ninest I also faced the same issue when I tried to test for my edit material endpoint. I looked up the other test code in the same test file, and realized for accessDeniedException test, the old test mocked the team that has no leads or heads. It is a weird error, so I think it's fine with what you did here to fix the test.

Reply:
yep had the same thing happen to me, had to do the same fix lol